driver-core/src/main/com/mongodb/client/model/expressions/MapExpression.java

```diff
@@ -21,45 +21,162 @@
+     * the key is present (which may be confirmed via {@link #has}). See
+     * {@link #get(StringExpression, Expression)} for a typesafe variant.
+     *
+     * @param key the key.
+     * @return the value.
+     */
     @MqlUnchecked(PRESENT)
     T get(StringExpression key);
 
-    // TODO-END doc "user asserts"
+    /**
+     * The value corresponding to the provided key.
+     *
+     * <p>Warning: The use of this method is an unchecked assertion that
+     * the key is present (which may be confirmed via {@link #has}). See
+     * {@link #get(StringExpression, Expression)} for a typesafe variant.
+     *
+     * @param key the key.
+     * @return the value.
+     */
     default T get(final String key) {
         return get(of(key));
     }
 
+    /**
+     * The value corresponding to the provided {@code key}, or the
+     * {@code other} value if an entry for the key is not present.
+     *
+     * @param key the key.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     T get(StringExpression key, T other);
 
+    /**
+     * The value corresponding to the provided {@code key}, or the
+     * {@code other} value if an entry for the key is not present.
+     *
+     * @param key the key.
+     * @param other the other value.
+     * @return the resulting value.
+     */
     default T get(final String key, final T other) {
         return get(of(key), other);
     }
 
+    /**
+     * Returns a map with the same entries as {@code this} map, but with
+     * the specified {@code key} set to the specified {@code value}.
+     *
+     * <p>This does not affect the original map.
+     *
+     * @param key the key.
+     * @param value the value.
+     * @return the resulting value.
+     */
     MapExpression<T> set(StringExpression key, T value);
 
+    /**
+     * Returns a map with the same entries as {@code this} map, but with
+     * the specified {@code key} set to the specified {@code value}.
+     *
+     * <p>This does not affect the original map.
+     *
+     * @param key the key.
+     * @param value the value.
+     * @return the resulting value.
+     */
     default MapExpression<T> set(final String key, final T value) {
         return set(of(key), value);
     }
 
+    /**
+     * Returns a map with the same entries as {@code this} map, but with
+     * the specified {@code key} absent.
+     *
+     * <p>This does not affect the original map.
+     *
+     * @param key the key.
+     * @return the resulting value.
+     */
     MapExpression<T> unset(StringExpression key);
 
+    /**
+     * Returns a map with the same entries as {@code this} map, but with
+     * the specified {@code key} absent.
+     *
+     * <p>This does not affect the original map.
+     *
+     * @param key the key.
+     * @return the resulting value.
+     */
     default MapExpression<T> unset(final String key) {
         return unset(of(key));
     }
```

Comment:
1. `absent` -> `{@linkplain #has(StringExpression) absent}`?
2. [optional] It appears that a single term should be chosen and used: either "has"/"does not have", or "present"/"not present", or "present"/"absent". But currently the documentation seemingly uses a random mix. Given that the method name is `has`, it would appear that "has"/"does not have" should be the terms. But maybe you will decide to rename the method `has`.